fix(web): Increase size of spacebar text

ermshiperete opened the pull request:
Fixes #7940.

The caption on the spacebar used to be 60% (0.6em). This change increases it to 100%.

One reason for the sometimes small text on the spacebar described in #7940 besides the too small font size on `kmw-spacebar-caption` was that the calculation of the ideal font size added double padding: it added two pixels of
padding to the height, but also multiplied the height value by 0.9 which effectively added more padding. This change removes the 2 pixel padding from the calculation.

Note that it is still possible for the text on the spacebar to become unreasonably small if the keyboard/language name is too long. However, that is outside of the scope of this PR.

![Screenshot from 2023-11-07 19-26-53](https://github.com/keymanapp/keyman/assets/181336/aae0c748-c8f5-4e03-881d-7bedbec408fa)

# User Testing

**TEST_EMULATOR**: 

- Go to the "Prediction - robust testing" test page in Chrome.
- Launch developer mode and set it to emulate an iPhone SE.
- Set the emulation scale to better match the relative size to a physical handheld iPhone SE.
- verify that the keyboard name on the spacebar has a reasonable size

**TEST_HARDWARE**:

- Install Keyman on an Android or iPhone device
- Open the Keyman app and install a Keyman keyboard
- verify that the keyboard name on the spacebar has a reasonable size

**TEST_WEB**:

- Go to the "Prediction - robust testing" test page in Chrome.
- verify that the keyboard name on the spacebar has a reasonable size

mcdurdin commented:
> This change increases it to 100%, but that might be sometimes too big. On the other hand, 80% still seemed to be too small.

I think we need to verify this across a range of keyboards and devices, because I expect that this is going to be too big for some of them. So we might need a cleverer solution, such as scaling it to fit?

mcdurdin commented:
I'm not also 100% sure that this is the entire cause of the tiny font size issue. The rest of the keyboard in the #7940 shows text at a sensible size, and it's only the spacebar which is mis-scaled. But all of the text elements on the keyboard use `em` scaling, so they should all be consistent in scale to each other?

ermshiperete commented:
Ah, you're right. It gets dynamically changed. Would have been too easy 😀.

ermshiperete commented:
> I think we need to verify this across a range of keyboards and devices, because I expect that this is going to be too big for some of them. So we might need a cleverer solution, such as scaling it to fit?

It already does downscaling so that the text will always fit in the available size. However, this can lead to the font size becoming unreasonably small if the language/keyboard name is too long, but that's a different issue.

ermshiperete commented:
> I'm not also 100% sure that this is the entire cause of the tiny font size issue. The rest of the keyboard in the #7940 shows text at a sensible size, and it's only the spacebar which is mis-scaled. But all of the text elements on the keyboard use `em` scaling, so they should all be consistent in scale to each other?

I added another commit to address the underlying cause of #7940 and updated the PR description.

mcdurdin commented on web/src/engine/osk/src/keyboard-layout/oskKey.ts:
```diff
@@ -239,14 +240,15 @@ export default abstract class OSKKey {
-    let textHeight = fontHeight ? fontHeight + Y_PADDING : 0;
+    // Don't add extra padding to height - multiplying with MAX_Y_PROPORTION already gives
+    // padding
+    let textHeight = fontHeight ? fontHeight : 0;
```
```suggestion
    let textHeight = fontHeight ?? 0;
```

bharanidharanj commented:
# Test Results

* **TEST_EMULATOR (PASSED):** Tested with the Keyman developer 17.0.207-alpha-test-9954 on Windows 10 OS and here is my observation: 1. Opened a Keyboard project. 2. Compiled it. Launched it in developer mode and set it to emulate an iPhone SE. 3. Verified that the keyboard name on the spacebar has a reasonable size.

..itransdevnagari keyboard

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/981730a4-e0cb-4b22-9771-d8b9b2cb112d">

..silcameroonqwerty

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/5392b230-2b14-4b81-be54-44bfe5954fae">
 
..khmerangkor

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/741c1097-876c-4cc5-8df9-7000d7136758">


* **TEST_HARDWARE (PASSED):** Installed PR build in the Android Mobile device (Samsung Galaxy A23 5g) and here is my observation: 1. Verified that the keyboard name on the spacebar has a reasonable size. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/57bbf38b-3086-403c-8ac9-8412617f6c92">


* **TEST_WEB (PASSED):** 1. Opened the "Prediction - robust testing" test page in Chrome browser. 2. Set it to emulation mode. 3. Verified that the keyboard name on the space has a reasonable size. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/e22e8297-133f-48e8-8e53-9344e33e2f63">

mcdurdin commented:
> * Launched it in developer mode and set it to emulate an iPhone SE.

@bharanidharanj, note that you need to reload the page after switching to Developer mode in Chrome (press Ctrl+R or F5). Otherwise, you get the desktop on screen keyboard in a phone form factor, which we can see is what has happened in your screenshots 😁.

@keymanapp-test-bot retest TEST_EMULATOR

bharanidharanj commented:
> > * Launched it in developer mode and set it to emulate an iPhone SE.
> 
> @bharanidharanj, note that you need to reload the page after switching to Developer mode in Chrome (press Ctrl+R or F5). Otherwise, you get the desktop on screen keyboard in a phone form factor, which we can see is what has happened in your screenshots 😁.
> 
> @keymanapp-test-bot retest TEST_EMULATOR

Thanks for the Clarification, @mcdurdin . I will retest and add my comment.

bharanidharanj commented:
# Test Results

* **TEST_EMULATOR (PASSED):** Retested as per Marc's suggestion and here is my observation: 1. Opened test page and set it to emulation mode in Chrome browser. 2. Verified that the keyboard name on the space has a reasonable size.

..English Keyboard

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/f82335f5-38e2-4989-bbcf-a87be157feaa">

..English - EuroLatin(SIL) Keyboard 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/d690f021-117c-4657-9edb-c9f02e55699f">

keyman-server commented:
Changes in this pull request will be available for download in [Keyman version 17.0.209-alpha](https://keyman.com/downloads/releases/alpha/17.0.209)

MayuraVerma commented:
Also the labels for alpha/number special keys, shift key, return key and global icon and text size should be increased 

![image](https://github.com/keymanapp/keyman/assets/14062593/2d20c1f0-6e1a-410a-8842-aff88301a925)

![image](https://github.com/keymanapp/keyman/assets/14062593/203d1833-e4c7-463a-aac0-41d6ee9db18a)

mcdurdin commented:
> Also the labels for alpha/number special keys, shift key, return key and global icon and text size should be increased

@MayuraVerma this needs to be a separate issue.
